src/Files.App/Helpers/UI/UIFilesystemHelpers.cs

```diff
@@ -324,7 +325,10 @@ public static async Task CreateFileFromDialogResultTypeAsync(AddItemDialogItemTy
 					break;
 			}
 
-			if (created.Status == ReturnResult.AccessUnauthorized)
+			// Add newly created item to recent files list
+			if (created.Status == ReturnResult.Success && created.Item?.Path is not null)
+				Shell32.SHAddToRecentDocs(Shell32.SHARD.SHARD_PATHW, created.Item.Path);
```

Comment:
this seems right! we basically do this when navigating to files/dirs in `NavigationHelper`

slight nitpick, but we could use `App.RecentItemsManager.AddToRecentItems(path)` to consolidate things

Reply:
Good idea